src/storage/fileio/unix/mod.rs

```diff
@@ -0,0 +1,195 @@
+use nix::libc;
+use std::fs::OpenOptions;
+use std::os::unix::prelude::FileExt;
+use std::{fs::File, io};
+
+use super::FileIO;
+
+pub struct UnixFileIO {}
```

Comment:
If this won't contain any data in the future, you can make this a unit struct with a semicolon and `#[repr(transparent)]`.

Reply:
Are you sure this is a proper use of repr(transparent)? https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent

Reply:
Alright, I tested the sizes and having `repr(transparent)` vs not having it for field-less structs doesn't make a difference. The compiler will optimize both to 0 size.